.github/workflows/publish-release.yml

```diff
@@ -33,14 +33,17 @@ jobs:
       - uses: MetaMask/action-publish-release@v3
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - uses: actions/cache@v3
+      - run: yarn --immutable
+      - run: yarn build
+      - name: Upload node modules and build artifacts
+        uses: actions/upload-artifact@v4
         with:
+          name: publish-release-artifacts-${{ github.sha }}
+          retention-days: 4
+          include-hidden-files: true
           path: |
             ./dist
-            ./node_modules/.yarn-state.yml
-          key: ${{ github.sha }}
-      - run: yarn --immutable
-      - run: yarn build
+            ./node_modules/
```

Comment:
We intentionally avoided running `yarn install` or including dependencies in the cache here in order to ensure third-party dependencies were not present during the step where the npm token is used, limiting risk of exfiltration.

Can we continue excluding it here? I am not following where it is needed

Reply:
It was indeed not intended. Looks like it was added here: https://github.com/MetaMask/metamask-module-template/pull/246
@mcmire I'm guessing that it was accidental to include third-party deps in the publish step, where they were previously excluded (I don't see this mentioned in the discussion)

Reply:
I think it was accidentally introduced, yes, sorry.